### ISSUE TITLE
Suggestion: Set Mangle Names vs. anonymous Context decls.

### ISSUE THREAD

**Issue by jonleivent** [coq/coq#13385]
Anonymous Context decls in a section or instance introduce names that are likely to clash with non-mangled hyp names, such as "H".  For instance:
```Coq
Set Mangled Names.
Section Foo.
  Context `{Bar}.
  Lemma baz : ....
    (*will see "H : Bar" as hypothesis*)
```
Furthermore, if class Bar has its own context parameters, these will also appear as non-mangled named hypotheses in interactive proofs in section `Foo` (or of proofs for instances of `Bar`).  I think it would be advantageous to mangle these names when `Mangled Names` is set so that they don't clash with explicit names used in interactive proofs.  This would make such proofs less fragile to cases where context parameters are added to the classes involved - such additions would not be local to the interactive proofs, or even to the section surrounding the proofs, but would introduce new unmangled hypothesis names that could clash which those used explicitly by the proofs.

Interestingly, Set Mangle Names currently has an effect on anonymous instances:
```Coq
Set Mangled Names.
Instance : Bar := .... (*instance will get a mangled name instead of Bar_instance_1*)
```
which doesn't seem useful as such instances don't clash with hypotheses.

**Comment by jashug** [comment]
Looks like maybe a duplicate of #13131, which was recently fixed in #13132 by @SkySkimmer

**Comment by Alizter** [comment]
This looks like it is working as intended in master, I've submitted a PR to add it to the test-suite and close this issue.
